### PR TITLE
Fix unit test failed due to localized culture and line ending (#7779)

### PR DESCRIPTION
**Pull request by lyndaidaii**
Summary of the changes:

* Change to Assert.StartWith instead of Assert.Equal, avoid checking localized 'parameter'  
* Add new class called 'UseInvariantCultureAttribute', change thread UI culture before test, and reset 
   after test
* Addressed those unit test:  1.NuGetGallery.Packaging.PackageMetadataFacts.ThrowsForEmptyAndNonEmptyDuplicatesWhenDuplicateMetadataElementsDetectedAndParsingIsNotStrict
2. NuGetGallery.Packaging.PackageMetadataFacts.ThrowsWhenInvalidMinClientVersion
3.NuGetGallery.PackagesControllerFacts+TheUploadFileActionForPostRequests.WillRejectBrokenZipFiles  NuGetGallery.PackagesControllerFacts+TheUploadFileActionForPostRequests.WillShowViewWithErrorsIfPackageIdIsBreaksParsing
4.NuGetGallery.Authentication.AuthenticationServiceFacts+TheAuthenticateMethod.GivenPasswordCredential_ItThrowsArgumentException
5.NuGetGallery.Diagnostics.DiagnosticsServiceFacts+TheGetSourceMethod.RequiresNonNullOrEmptyName
6.NuGetGallery.Services.ContentServiceFacts+TheGetContentItemMethod.GivenANullOrEmptyName_ItShouldThrow
 

Addresses https://github.com/NuGet/NuGetGallery/issues/123

### CODE REVIEW

File: tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using System.Globalization;
+using System.Reflection;
+using System.Threading;
+using Xunit.Sdk;
+
+namespace NuGetGallery.TestUtils
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class UseInvariantCultureAttribute : BeforeAfterTestAttribute
+    {
```
Comment (cristinamanum) [inline tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs:14]: This blog uses a similar idea https://hamidmosalla.com/2018/08/30/xunit-beforeaftertestattribute-how-to-run-code-before-and-after-test. Lazy initialization I think is good choice.

File: tests/NuGetGallery.Core.Facts/Packaging/PackageMetadataFacts.cs [hunk 2]
```diff
@@ -252,17 +255,23 @@ public void ThrowsPackagingExceptionWhenEmptyAndNonEmptyDuplicateMetadataElement
         public void ThrowsForEmptyAndNonEmptyDuplicatesWhenDuplicateMetadataElementsDetectedAndParsingIsNotStrict()
         {
             // Arrange
+            var useInvariantCultureAttr = new UseInvariantCultureAttribute();
             var packageStream = CreateTestPackageStreamWithDuplicateEmptyAndNonEmptyMetadataElements();
             var nupkg = new PackageArchiveReader(packageStream, leaveStreamOpen: false);
             var nuspec = nupkg.GetNuspecReader();
 
+            useInvariantCultureAttr.Before(null);
```
Comment (cristinamanum) [inline tests/NuGetGallery.Core.Facts/Packaging/PackageMetadataFacts.cs:263]: Can use the attribute on the test method like here: https://hamidmosalla.com/2018/08/30/xunit-beforeaftertestattribute-how-to-run-code-before-and-after-test/ ?

Comment (agr) [reply tests/NuGetGallery.Core.Facts/Packaging/PackageMetadataFacts.cs:263]: We definitely shouldn't instantiate the attribute class directly.

File: tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs (hunk 1 above)
Comment (agr) [reply tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs:14]: I'm not sure what benefits lazy initialization would bring in this particular case: we are not allocating anything at any point, just use static instance and keeping track of the original value.

Comment (lyndaidaii) [reply tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs:14]: I thought of those two ways Lazy initialization vs Static instance before, I think static instance works fine in this case.  I think Lazy initialization might be good if we are going to do more about culture in the future in terms of extensibility. Right now, I might prefer static instance? what do you guys think about those two approaches?

Comment (cristinamanum) [reply tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs:14]: I see, so use only InvariantCulture for now. It makes sense in this case to not have Lazy.

File: tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        private CultureInfo originalUICulture;
+
+        public CultureInfo OriginalUICulture { get; set; }
```
Comment (cristinamanum) [inline tests/NuGetGallery.Core.Facts/TestUtils/UseInvariantCultureAttribute.cs:17]: I am not sure if this is used anywhere, If  not can it be removed?

**Pull request merged**